packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -7,6 +7,10 @@ import { KeytipData } from '../../KeytipData';
 export interface ICheckboxState {
   /** Is true when Uncontrolled control is checked. */
   isChecked?: boolean;
+  isIndeterminate?: boolean;
```

Comment:
Why not match the name of the prop?
What happens if both `indeterminate` and `checked` props are enabled? Perhaps needs some details in documentation about that `indeterminate` wins priority initially and then it’s never updated even if they try. This will remove the need of the second state.

Reply:
I was just following suit with how the previous developer defined isChecked state as well as making it potentially easier to differentiate between props and state. I have currently reworked logic to make indeterminate state precede checked and defaultChecked, thanks!